### ISSUE TITLE
Any plans to updated Nuget Microsoft.OData.Edm 

### ISSUE THREAD

**Issue by toryb** [object/Simple.OData.Client#491]
Are there any plans to update the Nuget packages to utilize the latest version of the Microsoft.Data.*? I have a project where I want to utilize both Microsoft.Data.Client and Simple.OData.Client in the same project but they have dependencies to the same Nuget packages but differing versions (7.4.4 & 5.8.3 respectively). I am using functionality in Microsoft.Data.Client that requires later versions than the 5.8.3 but I am getting runtime errors if I try to use 7.4.4 with Simple.OData.Client.

**Comment by object** [comment]
Are you sure you are not mixing Microsoft.OData.* and Microsoft.Data.* packages? Microsoft OData packages comes in two versions:
- for OData V1-3 it's Microsoft.Data.OData and dependencies, last version 5.8.3.
- for OData V4 it's Microsoft.OData.Core and dependencies, last version 7.4.4.

Simple.OData.Client uses them both in order to support all OData protocol versions. If you only want subset, you can use Simple.OData.V3.Client and Simple.OData.V4 client that depend on Microsoft.Data.OData and Microsoft.OData.Core respectively.

All Simple.OData.Client packages are updated to use latest versions of Microsoft OData packages.

**Comment by toryb** [comment]
Thank you for the reply. As it turns out, the issue is related to using Microsoft.Extensions.Configuration.Json in the same project. When this package (and it's numerous dependencies are added to the project I get "System.MissingMethodException: 'Method not found: 'Void Simple.OData.Client.ODataClientSettings.set_BeforeRequest(System.Action`1<System.Net.Http.HttpRequestMessage>)'.'" 

I thought it was related to combining the multiple versions of Microsoft.Data.Edm & Microsoft.Data.Odata, etc. because I added the Microsoft.OData.Client NuGet package at the same time I added the Microsoft.Extensions.Configuration.Json. Sorry for the false diagnosis.

To recreate the issue, create a .NET library, include both the Simple.OData.V4.Client (or Simple.OData.Client) and the Microsoft.Extensions.Configuration.Json NuGet package (you don't even actually have to use it) and then create a unit test that calls a method in the library that attempts to assign a value for ODataClientSettings.BeforeRequest. You should get a "System.MissingMethodException".

Thanks.

**Comment by object** [comment]
Strange because Microsoft.Extensions.Configuration.Json doesn't have any common packages with Simple.OData.Client. Do you have a test project where this happens? Can you send it to me at vagif.abilov at gmail.com?

**Comment by toryb** [comment]
Yep - I'll send you a simple project.

**Comment by phatcher** [comment]
I had this a few weeks ago - I think its related to which version of HttpClient is found via the assembly binding redirects

**Comment by toryb** [comment]
@object I have attempted to send the project as a zip file and renamed it with a txt extension but it is rejected by gmail. "its content presents a potential security issue." So I have uploaded the solution zip here.

To reproduce the error, simply build the solution and run the single test case. If you debug the test case it won’t even step into the SSICRMSimpleODataDataSource.ConfigureClient() method.

As you can see from the code, I have commented out any code that uses Microsoft.Extensions.Configuration.Json. I have replaced the configuration data with hard-coded strings. To eliminate the exception, simply remove Microsoft.Extensions.Configuration.Json and its dependencies and recompile.

I have included the ODataConfig.json file for testing purposes. Because I did not include the app ID & app Secret, the expected behavior should be to fail authentication. With the Microsoft.Extensions.Configuration.Json package included the code will not get that far without raising the System.MissingMethodException exception.

Let me know if you need anything else. 

Thank you for looking into this.

[TestSimpleOData.zip](https://github.com/object/Simple.OData.Client/files/2141433/TestSimpleOData.zip)

**Comment by object** [comment]
Thank you @toryb I am on vacation until July 8 so not sure if I have time before I am back. But will check this out once I am back.

**Comment by toryb** [comment]
@object I have resolved my issue. I can't honestly say what was causing the issue because I just took this opportunity to migrate my entire framework (6 libraries plus test projects) to .net standard 2.0. This allowed (required) me to delete all the dependencies & references and all the binding redirections from the app.config files, etc. After migrating my target in my libraries from .net framework 4.6.2 to .net standard 2.0 and now having a very clean set of libraries I rebuilt all the dependencies using NuGet.

One thing I would suggest if someone runs into this issue (and doesn't want to do the migration I did) is to add the following property group in the test project if it is a .net framework project:
    <PropertyGroup>
        <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
        <GenerateBindingRedirectsOutputType>true</GenerateBindingRedirectsOutputType>
        <RestoreProjectStyle>PackageReference</RestoreProjectStyle>
    </PropertyGroup>

The first two items will make sure the *.config file that is generated will contain all the necessary redirects for the dependent libraries and the third line should make sure that the dependencies from any referenced libraries from a project in the solution will come with their dependent NuGet libraries.

So, after all my migration I am able to use the Simple.Odata.V4.Client with the Microsoft.Extensions.Configuration.Json library.
